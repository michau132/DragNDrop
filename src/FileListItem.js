import React from 'react';
import {
  TableCell, TableRow, Avatar, IconButton, Fade,
} from '@material-ui/core';
import { Delete } from '@material-ui/icons';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledTableCell = styled(TableCell)`
  && {
    padding: 4px 24px;
  }
`;

const StyledLastCell = styled(TableCell)`
  && {
    padding-left: 0;
  }
`;

const FileListItem = ({
  avatar, name, type, size, position = {}, removeImage, index,
}) => (
  <Fade in={!!name}>
    <TableRow>
      <StyledTableCell align="center"><Avatar src={avatar} /></StyledTableCell>
      <StyledTableCell align="center">{name}</StyledTableCell>
      <StyledTableCell align="center">{type}</StyledTableCell>
      <StyledTableCell align="center">{size}</StyledTableCell>
      <StyledTableCell>
      lat:
        {' '}
        {position.lat}
        <br />
      lng:
        {' '}
        {position.lng}
      </StyledTableCell>
      <StyledLastCell>
        <IconButton onClick={() => removeImage(index)}>
          <Delete />
        </IconButton>
      </StyledLastCell>
    </TableRow>
  </Fade>
);

FileListItem.propTypes = {
  avatar: PropTypes.string.isRequired,
  position: PropTypes.shape({
    lng: PropTypes.number.isRequired,
    lat: PropTypes.number.isRequired,
  }).isRequired,
  name: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  removeImage: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
};

export default FileListItem;
