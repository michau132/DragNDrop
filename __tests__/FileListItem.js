import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FileListItem from '../src/FileListItem';

const props = {
  avatar: 'someUrl',
  position: {
    lng: 12.12,
    lat: 12.13,
  },
  name: 'fileName.jpg',
  size: 12353,
  type: '.jpg',
  index: 0,
  removeImage: jest.fn(),
};

describe('<FileListItem/> component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FileListItem {...props} />);
  });
  test('should be defined', () => {
    expect(wrapper).toBeDefined();
  });
  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('should call removeImage', () => {
    wrapper.find('WithStyles(IconButton)').simulate('click');
    expect(props.removeImage).toHaveBeenCalledWith(0);
  });
});
