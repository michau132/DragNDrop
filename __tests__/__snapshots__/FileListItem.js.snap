// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileListItem/> component should match snapshot 1`] = `
<WithTheme(Fade)
  in={true}
>
  <WithStyles(TableRow)>
    <Styled(WithStyles(TableCell))
      align="center"
    >
      <WithStyles(Avatar)
        src="someUrl"
      />
    </Styled(WithStyles(TableCell))>
    <Styled(WithStyles(TableCell))
      align="center"
    >
      fileName.jpg
    </Styled(WithStyles(TableCell))>
    <Styled(WithStyles(TableCell))
      align="center"
    >
      .jpg
    </Styled(WithStyles(TableCell))>
    <Styled(WithStyles(TableCell))
      align="center"
    >
      12353
    </Styled(WithStyles(TableCell))>
    <Styled(WithStyles(TableCell))>
      lat:
       
      12.13
      <br />
      lng:
       
      12.12
    </Styled(WithStyles(TableCell))>
    <Styled(WithStyles(TableCell))>
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(DeleteIcon) />
      </WithStyles(IconButton)>
    </Styled(WithStyles(TableCell))>
  </WithStyles(TableRow)>
</WithTheme(Fade)>
`;
